crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/AppealContributionService.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.justice.laa.crime.contribution.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.justice.laa.crime.contribution.builder.AppealContributionResponseMapper;
+import uk.gov.justice.laa.crime.contribution.builder.CreateContributionRequestMapper;
+import uk.gov.justice.laa.crime.contribution.builder.GetContributionAmountRequestMapper;
+import uk.gov.justice.laa.crime.contribution.model.*;
+import uk.gov.justice.laa.crime.contribution.staticdata.enums.AssessmentResult;
+import uk.gov.justice.laa.crime.contribution.staticdata.enums.AssessmentStatus;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AppealContributionService {
+
+    private final MaatCourtDataService maatCourtDataService;
+    private final GetContributionAmountRequestMapper getContributionAmountRequestMapper;
+    private final CreateContributionRequestMapper createContributionRequestMapper;
+    private final AppealContributionResponseMapper appealContributionResponseMapper;
+
+    public AppealContributionResponse calculateContribution(AppealContributionRequest appealContributionRequest, String laaTransactionId) {
+        AssessmentResult assessmentResult = determineAssessmentResult(appealContributionRequest.getAssessments());
+
+        GetContributionAmountRequest getContributionAmountRequest = getContributionAmountRequestMapper.map(appealContributionRequest, assessmentResult);
+        BigDecimal appealContributionAmount = maatCourtDataService.getContributionAppealAmount(getContributionAmountRequest, laaTransactionId);
+
+        Integer repId = appealContributionRequest.getRepId();
+        Contribution currContribution =  maatCourtDataService.findContribution(repId, laaTransactionId);
+
+        if (!currContribution.getUpfrontContributions().equals(appealContributionAmount)) {
```

Comment:
Since you are checking BigDecimal for equality you should probably use `compareTo`

e.g. `boolean areSame = BigDecimal.ONE.compareTo(BigDecimal.ONE) == 0;`

Have a read through the JavaDoc for the `equals` and `compareTo` methods on BigDecimal for background.